src/factor.h

```diff
@@ -102,8 +121,8 @@ class CompactFactor {
                             : vif.satisfiedUsing(var_values[vif.vid]);
```

Comment:
It shouldn't but I believe there's no code that prevents it.  But on a second thought maybe we shouldn't do the check from a hot path.